src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -673,6 +676,71 @@ static int pcm_convert_s24_c32_to_s16_c32(const struct audio_stream *source,
 	return samples;
 }
 #endif
+
+#if CONFIG_PCM_CONVERTER_FORMAT_S24_C24_AND_S24_C32
+static int pcm_convert_s24_c24_to_s24_c32(const struct audio_stream *source,
+					  uint32_t ioffset, struct audio_stream *sink,
+					  uint32_t ooffset, uint32_t samples)
+{
+	uint8_t *src = source->r_ptr;
+	uint32_t *dst = sink->w_ptr;
+	int processed;
+	int nmax, i, n;
+
+	src += ioffset * 3;
+	dst += ooffset;
+	for (processed = 0; processed < samples; processed += n) {
+		src = audio_stream_wrap(source, src);
+		dst = audio_stream_wrap(sink, dst);
+		n = samples - processed;
+		nmax = audio_stream_bytes_without_wrap(source, src) / 3;
+		n = MIN(n, nmax);
+		nmax = audio_stream_bytes_without_wrap(sink, dst) >> BYTES_TO_S32_SAMPLES;
+		n = MIN(n, nmax);
+		for (i = 0; i < n; i += 1) {
+			*dst = (*(src + 2) << 16) | (*(src + 1) << 8) | *(src + 0);
+			dst++;
+			src += 3;
+		}
+	}
+
+	return samples;
+}
+
+static int pcm_convert_s24_c32_to_s24_c24(const struct audio_stream *source,
+					  uint32_t ioffset, struct audio_stream *sink,
+					  uint32_t ooffset, uint32_t samples)
+{
+	uint32_t *src = source->r_ptr;
+	uint8_t *dst = sink->w_ptr;
+	int processed;
+	int nmax, i, n;
+
+	src += ioffset;
+	dst += ooffset * 3;
+	for (processed = 0; processed < samples; processed += n) {
+		src = audio_stream_wrap(source, src);
+		dst = audio_stream_wrap(sink, dst);
+		n = samples - processed;
+		nmax = audio_stream_bytes_without_wrap(source, src) >> BYTES_TO_S32_SAMPLES;
+		n = MIN(n, nmax);
+		nmax = audio_stream_bytes_without_wrap(sink, dst) / 3;
+		n = MIN(n, nmax);
+		for (i = 0; i < n; i += 1) {
+			*dst = *src & 0xFF;
+			dst++;
+			*dst = (*src >> 8) & 0xFF;
+			dst++;
+			*dst = (*src >> 16) & 0xFF;
+			dst++;
+			src++;
+		}
```

Comment:
@singalsu any comments on the conversions ?

Reply:
@andrula-song we will need Hifi3/4 versions of these unless @mwasko already has some ?

Reply:
yes, I just have the same idea

Reply:
> @andrula-song we will need Hifi3/4 versions of these unless @mwasko already has some ?

I am focusing on the bug of PR https://github.com/thesofproject/sof/pull/5169 now, and I will start this after I finishing it.
and do we have testbench for this ?